sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3192,6 +3192,32 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
       checkAnswer(df3, Array(Row(new java.math.BigDecimal("0.100000000000000000000000100"))))
     }
   }
+
+  test("SPARK-29213: FilterExec should not throw NPE") {
+    withView("t1", "t2", "t3") {
+      sql("select ''").as[String].map(identity).toDF("x").createOrReplaceTempView("t1")
+      sql("select * from values 0, cast(null as bigint)")
+        .as[java.lang.Long]
+        .map(identity)
+        .toDF("x")
+        .createOrReplaceTempView("t2")
+      sql("select ''").as[String].map(identity).toDF("x").createOrReplaceTempView("t3")
+      sql(
+        """
+          |select t1.x
```

Comment:
nit: No strict rule though, I like capitalized words for SQL keywords: e.g., select -> SELECT, ....